File: web/src/api/activityEngine.js
import service from '@/utils/request'

// 活动管理
export const activityList = (data) => {
  return service({ url: '/activity_engine/activity/list', method: 'post', data })
}

export const activityAdd = (data) => {
  return service({ url: '/activity_engine/activity/add', method: 'post', data })
}

export const activityModify = (data) => {
  return service({ url: '/activity_engine/activity/modify', method: 'post', data })
}

export const activityDetail = (data) => {
  return service({ url: '/activity_engine/activity/detail', method: 'post', data })
}

export const activityPublish = (data) => {
  return service({ url: '/activity_engine/activity/publish', method: 'post', data })
}

export const activityOffline = (data) => {
  return service({ url: '/activity_engine/activity/offline', method: 'post', data })
}

// 模板管理
export const templateList = (data) => {
  return service({ url: '/activity_engine/template/list', method: 'post', data })
}

export const templateAdd = (data) => {
  return service({ url: '/activity_engine/template/add', method: 'post', data })
}

export const templateClone = (data) => {
  return service({ url: '/activity_engine/template/clone', method: 'post', data })
}

// 奖励道具
export const rewardItemSearch = (data) => {
  return service({ url: '/activity_engine/reward_item/search', method: 'post', data })
}

// 灰度管理
export const grayscaleUpdate = (data) => {
  return service({ url: '/activity_engine/grayscale/update', method: 'post', data })
}

// 沙箱测试
export const sandboxSimulate = (data) => {
  return service({ url: '/activity_engine/sandbox/simulate', method: 'post', data })
}

File: server/router/activity_engine/activity_engine.go
package activity_engine

import (
	"context"
	"sync"

	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/flipped-aurora/gin-vue-admin/server/router/activity_engine/initialize"
	"github.com/gin-gonic/gin"
)

type ActivityEngineRouter struct{}

var once sync.Once

func (r *ActivityEngineRouter) InitApiRouter(Router *gin.RouterGroup) {
	// 注册菜单和API（只执行一次）
	once.Do(func() {
		ctx := context.Background()
		initialize.Menu(ctx)
		initialize.Api(ctx)
	})

	apiRouter := Router.Group("activity_engine")
	apiRouter.Use(middleware.OperationRecord())
	{
		// 活动管理
		activityGroup := apiRouter.Group("activity")
		{
			activityGroup.POST("list", activityApi.List)
			activityGroup.POST("add", activityApi.Add)
			activityGroup.POST("modify", activityApi.Modify)
			activityGroup.POST("detail", activityApi.Detail)
			activityGroup.POST("publish", activityApi.Publish)
			activityGroup.POST("offline", activityApi.Offline)
		}

		// 模板管理
		templateGroup := apiRouter.Group("template")
		{
			templateGroup.POST("list", templateApi.List)
			templateGroup.POST("add", templateApi.Add)
			templateGroup.POST("clone", templateApi.Clone)
		}

		// 奖励道具
		rewardItemGroup := apiRouter.Group("reward_item")
		{
			rewardItemGroup.POST("search", rewardItemApi.Search)
		}

		// 灰度管理
		grayscaleGroup := apiRouter.Group("grayscale")
		{
			grayscaleGroup.POST("update", activityApi.UpdateGrayscale)
		}

		// 沙箱测试
		sandboxGroup := apiRouter.Group("sandbox")
		{
			sandboxGroup.POST("simulate", sandboxApi.Simulate)
		}
	}
}
